sugarcrm/client.py

```diff
@@ -0,0 +1,70 @@
+import urllib
+import json
+import hashlib
+import sugarcrm
+from sugarcrm.error import CrmAuthenticationError, CrmResponseError
+
+
+class APIClient:
+    def __init__(self, client=urllib.request, logger=sugarcrm.logger):
```

Comment:
`urllib.request` any particular reason not to use `requests` as HTTP client as we did in Koiki's client?

Reply:
I didn't have time to do the refactor since the original tests with the API I found was using urllib.parse.urlencode(args).encode("utf-8")  which I will have to translate to requests

I agree we should unify.